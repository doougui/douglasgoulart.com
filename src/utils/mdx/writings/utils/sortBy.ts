import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
