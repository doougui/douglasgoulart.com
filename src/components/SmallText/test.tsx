import { render, screen } from 'utils/tests';

import { SmallText } from '.';

describe('<SmallText />', () => {
  it('should render correctly', () => {
    const { container } = render(<SmallText>This is a small text</SmallText>);

    expect(screen.getByText('This is a small text')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
