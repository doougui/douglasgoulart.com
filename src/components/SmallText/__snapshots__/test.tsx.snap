// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallText /> should render correctly 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  color: #888888;
  margin-top: 2px;
}

<small
  class="c0"
>
  This is a small text
</small>
`;
